clients/command-amqp/src/main/java/org/eclipse/hono/client/command/amqp/ProtonBasedInternalCommandSender.java

```diff
@@ -93,12 +95,10 @@ public ProtonBasedInternalCommandSender(final HonoConnection connection) {
                     }
                     return (Void) null;
                 })
-                .recover(thr -> {
+                .onFailure(thr -> {
                     LOG.debug("failed to send command [{}] to downstream peer", commandContext.getCommand(), thr);
-                    TracingHelper.logError(commandContext.getTracingSpan(),
-                            "failed to send command message to downstream peer: " + thr);
-                    commandContext.release(thr);
-                    return Future.succeededFuture();
+                    commandContext.release(new ServerErrorException(HttpURLConnection.HTTP_UNAVAILABLE,
+                            "failed to send command message to downstream peer: " + thr));
```

Comment:
FMPOV the *thr* should be included as the root cause here instead of its String serialization...

Reply:
That would prevent the `thr` root cause to be logged because the `error.object` span log field only contains the `exception.toString` representation.

Reply:
Then we should probably change that in *TracingHelper.logError()*, right?

Reply:
That would mean adding an extra (non-standard) `error.cause.object` span log item, I guess.

I don't think that we need this in most cases where a Throwable with a non-null `cause` is given to `TracingHelper.logError()`.
Oftentimes the nested exception will have been logged in a child span already anyway.
Or it could be considered too low-level to include in tracing output (we are also not including stacktraces there).

So, I would rather say that if for a particular span the nested exception message should be logged, it should be done by adding an explicit `TracingHelper.logError` invocation.

Reply:
> That would mean adding an extra (non-standard) error.cause.object span log item, I guess.

Sounds good :+1:

Reply:
> I don't think that we need this in most cases where a Throwable with a non-null cause is given to TracingHelper.logError().

In most cases we do not specify a root cause anyway. So there shouldn't be a problem logging it using the *error.cause.object* field in the rare cases where we do set it, right?

Reply:
I've had a look here and yes, in most cases there's no root cause given.
And there are quite some cases, where logging the nested exception would indeed be beneficial - e.g. for an exception like `new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST, e)`.

So, I've made the corresponding changes now, adding a `error.cause.object` field.